### PR TITLE
Use skimage map array

### PR DESCRIPTION
**Pull request by zoccoler**
This is an initial attempt to use [skimage `map_array`](https://scikit-image.org/docs/stable/api/skimage.util.html#map-array) instead of [`numpy.take`](https://numpy.org/doc/stable/reference/generated/numpy.take.html) and possibly `pyclesperanto.replace_intensities`.

This can potentially solve #65, #59 and #48 .

## To do:
- [x] write tests

### CODE REVIEW

**Comment by zoccoler** [comment]
Hi @haesleinhuepf ,

Please have a look at this when you get some time and let me know what you think!

Best,
Marcelo

**Review by haesleinhuepf — CHANGES_REQUESTED** [review]
Hi Marcelo @zoccoler ,

thanks for spending the time! I would prefer having a solution without `use_map_array=False` parameters. See my comment [here](https://github.com/haesleinhuepf/napari-skimage-regionprops/issues/48#issue-1556700890). We should not modify existing function. We should make new ones that work better and deprecate the old ones. Adding parameters such as `do_this` and `do_that` will ultimately lead to long spaghetti-code that is hard to maintain. Add new (simple) functions, deprecating the old and at some point removing them is the better strategy long-term. Happy to discuss in person if it's not clear.

Thanks!

**Comment by zoccoler** [comment]
Hi Robert @haesleinhuepf ,

Thank you very much for your hints ! It helps me learn the proper way to deprecate old functions.
Here is my new attempt to do that by creating new functions and deprecating old ones.

Please let me know if you think this can be further simplified or changed in case it is not the proper way :)

File: napari_skimage_regionprops/_parametric_images.py [hunk 1]
```diff
@@ -1,7 +1,87 @@
 import numpy as np
 from napari_tools_menu import register_function
 import numpy
+from deprecated import deprecated
+from napari.utils import notifications
```
Comment (haesleinhuepf) [inline napari_skimage_regionprops/_parametric_images.py:5]: Can you please delay this import? (Import it from within the function where you use it)

File: napari_skimage_regionprops/_parametric_images.py [hunk 2]
```diff
@@ -1,7 +1,87 @@
...
 
+@register_function(menu="Measurement maps > Measurements on labels with map array (nsr)")
+@register_function(menu="Visualization > Measurements on labels with map array (nsr)")
+def visualize_measurement_on_labels_with_map_array(labels_layer:"napari.layers.Labels", column:str = "label", viewer:"napari.Viewer" = None) -> "napari.types.ImageData":
```
Comment (haesleinhuepf) [inline napari_skimage_regionprops/_parametric_images.py:9]: The name of the function is a bit cumbersome. How about `map_measurements_on_labels`?

File: napari_skimage_regionprops/_parametric_images.py [hunk 3]
```diff
@@ -1,7 +1,87 @@
...
+    measurements = np.asarray(table_one_timepoint[column]).tolist()
+    return relabel_skimage(labels_one_timepoint, measurements)
+
+def relabel_skimage(image, measurements):
+    from skimage.util import map_array
+    return map_array(image, np.unique(image), np.array(measurements))
+
+@deprecated("This function is deprecated! To adhere to future behavior and suppress this warning, use 'visualize_measurement_on_labels_with_map_array' instead")
 @register_function(menu="Measurement maps > Measurements on labels (nsr)")
 @register_function(menu="Visualization > Measurements on labels (nsr)")
```
Comment (haesleinhuepf) [inline napari_skimage_regionprops/_parametric_images.py:86]: As we don't want people to continue using this function, and we have a replacement, we can remove it from the menu.

**Review by haesleinhuepf — COMMENTED** [review]
Hi Marcelo @zoccoler ,

great! This looks very good (minor comments below). 

Thanks!

Best,
Robert

File: napari_skimage_regionprops/_parametric_images.py (hunk 1 above)
Comment (zoccoler) [reply napari_skimage_regionprops/_parametric_images.py:5]: Sure! Good that you remember me, thanks!

File: napari_skimage_regionprops/_parametric_images.py (hunk 2 above)
Comment (zoccoler) [reply napari_skimage_regionprops/_parametric_images.py:9]: I love it! 😻

File: napari_skimage_regionprops/_parametric_images.py (hunk 3 above)
Comment (zoccoler) [reply napari_skimage_regionprops/_parametric_images.py:86]: OK, done!

**Review by zoccoler — COMMENTED** [review]
Fixed changes

**Pull request merged**